common-api/src/main/scala/pl/touk/nussknacker/engine/graph/expression/TabularTypedData.scala

```diff
@@ -122,14 +146,28 @@ object TabularTypedData {
 
   implicit class Stringify(data: TabularTypedData) {
 
-    def stringify: String = Coders.TabularTypedDataEncoder(data).noSpaces
+    def stringify: String = Coders
+      .TabularTypedDataEncoder(
+        (data.columnDefinitions, data.rows.map(_.cells.map(_.rawValue)))
+      )
+      .noSpaces
+
   }
 
-  def fromString(value: String): Either[circe.Error, TabularTypedData] = {
+  def fromString(value: String): Either[Error, TabularTypedData] = {
     for {
-      json <- io.circe.parser.parse(value)
-      data <- Coders.TabularTypedDataDecoder.decodeJson(json)
-    } yield data
+      json <- io.circe.parser.parse(value).left.map(failure => CannotParseError(failure.message))
+      data <- Coders.TabularTypedDataDecoder.decodeJson(json).left.map(failure => CannotParseError(failure.message))
+      (columns, rows) = data
+      tabularTypedData <- TabularTypedData.create(columns, rows).left.map(CannotCreateError.apply)
+    } yield tabularTypedData
+  }
+
+  sealed trait Error
+
+  object Error {
+    final case class CannotParseError(message: String)       extends Error
```

Comment:
The `parse` vs `create` parts are a little bit mysterious. I think that we can make it more explicit.
1. WDYT about `CannotParseError` -> `JsonParsingError`/`JsonParseError`?
2. For the `create` part I have alternatives like `TabularValidationError`/`TabularDataValidationError`